bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -516,6 +517,31 @@ impl Room {
     pub async fn typing_notice(&self, is_typing: bool) -> Result<(), ClientError> {
         Ok(self.inner.typing_notice(is_typing).await?)
     }
+
+    /// Sets a flag on the room to indicate that the user has explicitly marked it as unread
+    pub async fn mark_as_unread(&self) -> Result<(), ClientError> {
+        Ok(self.inner.mark_unread(true).await?)
+    }
+
+    /// Reverts a previously set unread flag.
+    /// Optionally sends a read receipt to the latest event in the room.
+    /// Sending read receipts is useful when executing this from the room list
+    /// but shouldn't be use when entering the room, the timeline should be left to
+    /// its own devices in that case.
+    pub async fn mark_as_read(
+        &self,
+        send_read_receipt: bool,
+        receipt_type: ReceiptType,
+    ) -> Result<(), ClientError> {
+        if send_read_receipt {
+            let timeline = self.timeline().await?;
+            if let Some(event) = timeline.latest_event().await {
+                let _ = timeline.send_read_receipt(receipt_type, event.event_id().unwrap());
+            }
+        }
+
+        Ok(self.inner.mark_unread(false).await?)
+    }
```

Comment:
Hmmmm 🤔. I wonder if we shouldn't have `mark_as_read` along with another method: `mark_as_read_and_send_a_read_receipt`. Alternatively, if it's already possible to send a read receipt manually, then let's keep `mark_as_read` alone, and use the other method to simulate the `send_read_receipt` argument. Thoughts?

Reply:
We have a `send_read_receipt` method but it doesn't automatically fetch the `latest_event` from the timeline, it relies on it being passed from the outside. Basically you don't like the `send_read_receipt` parameter and would like to make that explicit in the function name, right?

Reply:
Yes exactly.